lambdas/checkpassport/src/main/java/uk/gov/di/ipv/cri/passport/checkpassport/CheckPassportHandler.java

```diff
@@ -116,24 +115,32 @@ public APIGatewayProxyResponseEvent handleRequest(
         try {
             String passportSessionId = RequestHelper.getPassportSessionId(input);
 
-            passportSessionService.incrementAttemptCount(passportSessionId);
-
-            Map<String, List<String>> queryStringParameters = getQueryStringParametersAsMap(input);
-            String userId = RequestHelper.getHeaderByKey(input.getHeaders(), "user_id");
+            PassportSessionItem passportSessionItem =
+                    passportSessionService.getPassportSession(passportSessionId);
 
-            var validationResult =
-                    authRequestValidator.validateRequest(queryStringParameters, userId);
-            if (validationResult.isPresent()) {
+            if (passportSessionItem == null) {
                 return ApiGatewayResponseGenerator.proxyJsonResponse(
                         HttpStatus.SC_BAD_REQUEST,
                         new ErrorObject(
                                         OAuth2Error.SERVER_ERROR_CODE,
-                                        validationResult.get().getMessage())
+                                        ErrorResponse.PASSPORT_SESSION_NOT_FOUND.getMessage())
                                 .toJSONObject());
             }
 
+            passportSessionService.incrementAttemptCount(passportSessionId);
+
+            String userId = passportSessionItem.getUserId();
+            var authParams = passportSessionItem.getAuthParams();
+
             AuthenticationRequest authenticationRequest =
-                    AuthenticationRequest.parse(queryStringParameters);
+                    new AuthenticationRequest(
```

Comment:
I'm not 100 sure now if this AuthenticationRequest is the correct thing we need to use. The fact that it forces you to have Scope("openid") probably means this is for an OIDC authentication endpoint which isn't quite what we want.

I know this this what we had before so we don't need to fix it here, but this is something we probably want to address and fix.

Reply:
Changed to AuthorizationRequest